lib/TPP/Dialect/VNNI/TransformOps/VNNITransformOps.cpp

```diff
@@ -0,0 +1,88 @@
+//===- VNNITransformOps.cpp - Implementation of VNNI transform ops--====//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Dialect/VNNI/TransformOps/VNNITransformOps.h"
+#include "TPP/Dialect/Tpp/TppOps.h"
+#include "TPP/Dialect/VNNI/VNNIOps.h"
+#include "TPP/Transforms.h"
+#include "mlir/AsmParser/AsmParser.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Transform/IR/TransformDialect.h"
+#include "mlir/Dialect/Transform/IR/TransformInterfaces.h"
+#include "mlir/Interfaces/ViewLikeInterface.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Casting.h"
+
+using namespace mlir;
+using namespace mlir::transform;
+using namespace mlir::tpp;
+//===----------------------------------------------------------------------===//
+// Transform op registration
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+class VNNITransformDialectExtension
+    : public transform::TransformDialectExtension<
+          VNNITransformDialectExtension> {
+public:
+  using Base::Base;
+
+  void init() {
+    registerTransformOps<
+#define GET_OP_LIST
+#include "TPP/Dialect/VNNI/TransformOps/VNNITransformOps.cpp.inc"
+        >();
+  }
+};
+} // namespace
+
+namespace {
+// A simple pattern rewriter that implements no special logic.
+class SimpleRewriter : public PatternRewriter {
+public:
+  SimpleRewriter(MLIRContext *context) : PatternRewriter(context) {}
+};
+} // namespace
+
+DiagnosedSilenceableFailure
+transform::MapVNNIToTppOp::applyToOne(Operation *target,
+                                      SmallVector<Operation *> &results,
+                                      TransformState &state) {
+  if (auto matmulOp = dyn_cast<vnni::MatmulOp>(target)) {
```

Comment:
can you please return failure immediately instead of nesting inside an if, thanks